src/tests/JIT/Directed/tailcall/mutual_recursion.fs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// Notes:
+//
+// This test will Stack overflow when built in debug. It relies on the FSharp
+// compiler optimizing the tail calls to loops.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+open System
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+// This is a 16 byte struct
+type Point2D = struct
+    val X: double
+    val Y: double
+
+    new(x: double, y:double) = { X = x; Y = y}
+
+    end
```

Comment:
More style than anything else, it's more common now in F# to use the attribute and primary constructor:

```suggestion
[<Struct>]
type Point2D(x: double, y: double) =
    member _.X = x
    member _.Y = y
```